### PR TITLE
PE-64 | Created a Service for Profile Onboarding

### PR DESCRIPTION
**Pull request by tonynguyen98**
**What I Did:**
- Created a HTTP POST service file that will add new users.
- Created a HTTP PUT service file that will update user. However this currently does not work until I can get the current user_id, which is another task done by @davidsquines.
- Reworked the routing so the Profile Onboarding Screen will show if the user is a first time account.

**How to Test:**
1. Delete your account from Firebase Auth.
2. Create a new account.
3. The Profile Onboarding should pop up, fill it out accordingly. 
4. Now check MongoDB for the Users Table to see if the values you've added have been added properly.
5. Now try step 3 and 4 again with a Google Account instead.

### CODE REVIEW

File: lib/screens/profile_onboarding.dart [hunk 1]
```diff
@@ -162,26 +166,56 @@ class _ProfileOnboardingState extends State<ProfileOnboarding> {
...
+        Navigator.pop(context);
+      } else {
+        // Using HTTP POST to add new user
+        await _profileOnboardingService.addUser(user);
+
+        Navigator.pushAndRemoveUntil(
+          context,
+          MaterialPageRoute(
+            builder: (context) => Home(),
+          ),
```
Comment (ayeejane) [inline lib/screens/profile_onboarding.dart:212]: When I click the `Confirm` button nothing happens but when I hot reload, it takes me to the home screen.

**Review by ayeejane — CHANGES_REQUESTED** [review]

File: lib/screens/profile_onboarding.dart [hunk 2]
```diff
@@ -162,26 +166,56 @@ class _ProfileOnboardingState extends State<ProfileOnboarding> {
...
+      lastName: _lastName,
+      firebaseUuid: _firebaseUser.uid,
+      email: _firebaseUser.email,
+      city: _city,
+      category: [], //we don't have form field for this
+      pictureUrl: '',
+      role: _role,
+      ratings: [],
+      averageRating: 0.0,
+    );
```
Comment (txt020) [inline lib/screens/profile_onboarding.dart:194]: For create profile. The Months and Day and Year do not indicate what this information is for.

Is it my birthday or the day I joined this app?
image

File: lib/screens/profile_onboarding.dart (hunk 1 above)
Comment (txt020) [reply lib/screens/profile_onboarding.dart:212]: I am having the same issue as well. When I fill out all the information and click Confirm I am not routed to home screen.

**Review by txt020 — CHANGES_REQUESTED** [review]

File: lib/screens/profile_onboarding.dart (hunk 2 above)
Comment (tonynguyen98) [reply lib/screens/profile_onboarding.dart:194]: I am currently considering removing that field altogether. We don't have a field for it in our back-end, and our front end has no use for it to be honest. I will discuss this during the next meeting.

File: lib/screens/profile_onboarding.dart (hunk 1 above)
Comment (tonynguyen98) [reply lib/screens/profile_onboarding.dart:212]: I just pushed a new commit that hopefully will try to fix the problem both of you are having, can you guys confirm?

**Review by ayeejane — APPROVED** [review]
Works as intended

**Review by txt020 — APPROVED** [review]
Looks Good! 
The .env bug was fixed.

**Review by dominikyz — APPROVED** [review]
Works perfect for both email and gmail

**Pull request merged**